pkg/prowgen/prowgen.go

```diff
@@ -454,21 +427,40 @@ func generatePeriodicForTest(name string, info *ProwgenInfo, label jc.ProwgenLab
 	}
 }
 
-func generateClusterProfileVolume(name, profile string) corev1.Volume {
-	return corev1.Volume{
-		Name: name,
+func generateClusterProfileVolume(profile cioperatorapi.ClusterProfile, clusterType string) corev1.Volume {
+	ret := corev1.Volume{
+		Name: "cluster-profile",
 		VolumeSource: corev1.VolumeSource{
 			Projected: &corev1.ProjectedVolumeSource{
 				Sources: []corev1.VolumeProjection{{
 					Secret: &corev1.SecretProjection{
 						LocalObjectReference: corev1.LocalObjectReference{
-							Name: profile,
+							Name: fmt.Sprintf("cluster-secrets-%s", clusterType),
 						},
 					}},
 				},
 			},
 		},
 	}
+	switch profile {
+	case cioperatorapi.ClusterProfileAWS,
+		cioperatorapi.ClusterProfileAzure4,
+		cioperatorapi.ClusterProfileLibvirtS390x,
+		cioperatorapi.ClusterProfileOpenStack,
+		cioperatorapi.ClusterProfileOpenStackVexxhost,
+		cioperatorapi.ClusterProfileOpenStackPpc64le,
+		cioperatorapi.ClusterProfileVSphere:
+	default:
+		s := &ret.VolumeSource.Projected.Sources
+		*s = append(*s, corev1.VolumeProjection{
```

Comment:
just `append(ret.VolumeSource.Projected.Sources, corev1.VolumeProjection{` is the same as getting a pointer, then instantly dereferencing it or not?

Reply:
Yeah, `ret.VolumeSource.Projected.Sources = append(ret.VolumeSource.Projected.Sources, ...)` would be the same